### PR TITLE
Added new alpha: Dual Thrust model and sample algorithm.

### PR DESCRIPTION
**Pull request by IlshatGaripov**
#### Description
The published file contains new alpha model along with an algorithm.
Alpha model implements a widely known concept to generate trading singnals called Dual Thrust. https://www.quantconnect.com/tutorials/strategy-library/dual-thrust-trading-algorithm
As a demonstration example trading algorithm that utilize an alpha model has been created.
Traded asset is UVXY. With current input parameters set an algorithm shows a theoretical profit on a backtest from Oct 2016 till Jan 2019 of around 400 percent. Possibly could be run on realtime data for demonstration purposes.

#### Related Issue
NA

#### Motivation and Context
Add more alphas to an opensource. This model implements the basic idea and can be use in conjunction with other alpha models.

#### Requires Documentation Change
NA

#### How Has This Been Tested?
I have run it locally on custom data. It does work fine. No errors.

#### Types of changes
* [ ]   Bug fix (non-breaking change which fixes an issue)
* [X ] New feature (non-breaking change which adds functionality)
* [ ]   Breaking change (fix or feature that would cause existing functionality to change)
* [ ]   Non-functional change (xml comments/documentation/etc)

#### Checklist:
* [ X] My code follows the code style of this project.
* [ X] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
* [ ]   I have added tests to cover my changes.
* [ ]   All new and existing tests passed.
* [X ] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm.CSharp/Alphas/VixDualThrustAlphaAlgorithm.cs [hunk 1]
```diff
@@ -0,0 +1,265 @@
+﻿using System;
```
Comment (jaredbroad) [inline Algorithm.CSharp/Alphas/VixDualThrustAlphaAlgorithm.cs]: Please add license header here

File: Algorithm.CSharp/Alphas/VixDualThrustAlphaAlgorithm.cs [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            SetAlpha(new DualThrustAlphaModel(_k1, _k2, _rangePeriod, UniverseSettings.Resolution, _consolidatorBars));
+
+            // Portfolio Construction
+            SetPortfolioConstruction(new EqualWeightingPortfolioConstructionModel());
+
+            // Execution
+            SetExecution(new ImmediateExecutionModel());
+
+            // Risk Management
+            SetRiskManagement(new MaximumDrawdownPercentPerSecurity(0.03m));
```
Comment (jaredbroad) [inline Algorithm.CSharp/Alphas/VixDualThrustAlphaAlgorithm.cs]: Please use warm up or history so the SymbolData instantly ready on deployment.

File: Algorithm.CSharp/Alphas/VixDualThrustAlphaAlgorithm.cs [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        private decimal _k2 = 0.63m;
+        private int _rangePeriod = 20;
+        private int _consolidatorBars = 30;
+
+        // -- INITIALIZE --
+        public override void Initialize()
+        {
+            // Set Backtesting Period
+            SetStartDate(2016, 10, 01);
+            SetEndDate(2019, 01, 01);
```
Comment (jaredbroad) [inline Algorithm.CSharp/Alphas/VixDualThrustAlphaAlgorithm.cs]: Please remove end date and set 0 fee model for alphas

File: Algorithm.CSharp/Alphas/VixDualThrustAlphaAlgorithm.cs [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+using QuantConnect.Brokerages;
+using QuantConnect.Data;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Data.Market;
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Indicators;
+
+namespace QuantConnect.Algorithm.CSharp.Alphas
+{
+    class VIXDualThrustAlphaAlgorithm : QCAlgorithmFramework
```
Comment (jaredbroad) [inline Algorithm.CSharp/Alphas/VixDualThrustAlphaAlgorithm.cs]: Please rename to end in Alpha, pedantic but keeps the pattern of the other alphas

**Comment by IlshatGaripov** [comment]
Thank for remarks! I have tried to make the requested changed and have re submitted. Please kindly check again if things are in a right place ?

**Pull request merged**